src/network/esp/src/NetworkConfig.cpp

```diff
@@ -0,0 +1,23 @@
+#include "NetworkConfig.h"
+
+#include "DefaultNetworkConfig.h"
+#include "esp_event.h"
+#include "esp_log.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include <cstring>
+
+wifi_mode_t NetworkConfig::getMode() { return WIFI_MODE_STA; }
+
+wifi_config_t* NetworkConfig::getDefaultNetworkConfig() {
+    static wifi_config_t s_wifiConfig;
+    std::memcpy(s_wifiConfig.sta.ssid, DEFAULT_SSID, strlen(DEFAULT_SSID));
+    std::memcpy(s_wifiConfig.sta.password, DEFAULT_PASSWORD, strlen(DEFAULT_PASSWORD));
```

Comment:
use strcpy, it avoids parsing the string for the length, then again to copy the bytes.
If you know the max length of password and ssid, you could use snprintf so you don't write out of bounds just to be safe
` snprintf(buf, maxlength, "%s", cstr);`